TheModernIlluminati/TheModernIlluminati/Pages/SearchCountry.cshtml.cs

```diff
@@ -41,21 +45,28 @@ public void OnPost()
                 IDictionary<long, TheModernIlluminati.Models.Nobel> allNobels = new Dictionary<long, TheModernIlluminati.Models.Nobel>();
                 string nobelJSON = webClient.DownloadString("http://api.nobelprize.org/v1/laureate.json");
                 TheModernIlluminati.Models.Nobel nobel = TheModernIlluminati.Models.Nobel.FromJson(nobelJSON);
-                List<TheModernIlluminati.Models.Laureate> laureate1 = nobel.Laureates;
-                List<TheModernIlluminati.Models.Laureate> laureate2 = new List<TheModernIlluminati.Models.Laureate>();
+                List<TheModernIlluminati.Models.Laureate> nobelLaureates = nobel.Laureates;
+                List<TheModernIlluminati.Models.Laureate> laureates = new List<TheModernIlluminati.Models.Laureate>();
                
-                foreach (var laureate4 in laureate1)
+                foreach (var laureate in nobelLaureates)
                 {
-                    if (laureate4.BornCountryCode == CountryCode)
+                    if (laureate.BornCountryCode == CountryCode && laureate.Gender.ToString()!= "Org")
                     {
-                        laureate2.Add(laureate4);
+                       
+                        laureates.Add(laureate);
                     }
 
                 }
-                ViewData["filteredLaureate"] = laureate2;
+
+                if (laureates.Count() != 0)
+                {
+                    ViewData["filteredLaureate"] = laureates;
+                }
+               
 
             }
-            searchFinished = true;
+            
+            return Page();
```

Comment:
Added return Page() in OnPost.

source : https://stackoverflow.com/questions/50567901/what-does-page-method-in-razor-pages-onpost-method-do